addons/point_of_sale/static/src/app/screens/product_screen/product_screen.js

```diff
@@ -496,6 +496,18 @@ export class ProductScreen extends Component {
     }
 
     async addProductToOrder(product) {
+        if (this.searchWord && product.isConfigurable()) {
+            const barcode = this.searchWord;
+            const searchedProduct = this.pos.models["product.product"].filter(
+                (p) =>
+                    p.raw.product_tmpl_id === product.raw.product_tmpl_id &&
+                    p.barcode &&
+                    p.barcode.includes(barcode)
+            );
+            if (searchedProduct.length === 1) {
+                product = searchedProduct[0];
+            }
```

Comment:
```suggestion
            const barcode = this.searchWord;
            const templateId = product.raw.product_tmpl_id;
            const searchedProduct = this.pos.models["product.product"].getBy("barcode", barcode);
            if (searchedProduct && searchedProduct.raw.product_tmpl_id === templateId) {
                product = searchedProduct[0];
            }
```

I think we can afford to add auto-completion of attributes only when the barcode is strictly equal to the search. 

This will be more efficient, given that the index already exists.